x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/DownsampleStep.java

```diff
@@ -34,21 +35,24 @@ public class DownsampleStep extends AsyncActionStep {
     private static final Logger logger = LogManager.getLogger(DownsampleStep.class);
 
     private final DateHistogramInterval fixedInterval;
+    private final TimeValue waitTimeout;
     private final StepKey nextStepOnSuccess;
     private final StepKey nextStepOnFailure;
     private volatile boolean downsampleFailed;
 
     public DownsampleStep(
-        StepKey key,
-        StepKey nextStepOnSuccess,
-        StepKey nextStepOnFailure,
-        Client client,
-        DateHistogramInterval fixedInterval
+        final StepKey key,
+        final StepKey nextStepOnSuccess,
+        final StepKey nextStepOnFailure,
```

Comment:
ah one more thing, I believe the `DownsampleStep` doesn't need this dance to go back to the `clean` step. It can now be just a regular `AsyncActionStep` with one `nextStepKey` and whenever it fails it'll go into the `ERROR` step and be retried automatically later (now, the downsample transport will support retries without an issue)

See https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/RolloverStep.java#L34 as an example of a regular async step